src/coap_blockwise.h

```diff
@@ -26,6 +28,24 @@ enum golioth_status golioth_blockwise_post(struct golioth_client *client,
                                            golioth_set_cb_fn callback,
                                            void *callback_arg);
 
+/* Blockwise Multi-Part Upload */
+
+// Note: callback_arg will be the same for all blocks
+struct blockwise_transfer *golioth_blockwise_upload_start(const char *path_prefix,
+                                                          const char *path,
+                                                          enum golioth_content_type content_type,
+                                                          golioth_set_cb_fn cb,
+                                                          void *callback_arg);
+
+void golioth_blockwise_upload_stop(struct blockwise_transfer *ctx);
+
+enum golioth_status golioth_blockwise_upload_block(struct golioth_client *client,
+                                                   struct blockwise_transfer *ctx,
+                                                   uint32_t block_idx,
+                                                   uint8_t *block_buffer,
+                                                   size_t data_len,
```

Comment:
Length arguments should match the naming of the corresponding pointer argument, for clarity.
```suggestion
                                                   size_t block_len,
```